### DISCUSSION TITLE
What's the status of the session options config keys?

### DISCUSSION THREAD

**Discussion by Craigacp** [microsoft/onnxruntime#7064] [Other Q&A]
I was looking through https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h and wondering about the status of those strings. Is it for things that are unstable (as otherwise they'd have actual methods in the C API), or is it to reduce bloat in the C API?

Should bindings like Java provide proper method entry points which supply the string themselves and enforce type & value safety? E.g. only allow the load_model_format option to be set with the string `ORT`, or unset completely.

**Comment by pranavsharma** [comment]
It's to reduce bloat in the C API.

**Comment by Craigacp** [comment]
So we should provide proper type safe bindings in higher level APIs then? Handling the type safety in Java will be simple as it will all go through the method that passes a key value pair to the native code, but we can guarantee that nothing else will get passed in.
